frontend/src/metabase/common/components/Table/ControlledTable.tsx

```diff
@@ -0,0 +1,144 @@
+import React from "react";
+
+import { PaginationControls } from "metabase/components/PaginationControls";
+import { color } from "metabase/lib/colors";
+import { Box, Flex, Icon } from "metabase/ui";
+
+import CS from "./Table.module.css";
+
+export type BaseRow = Record<string, unknown> & { id: number | string };
+
+type ColumnItem = {
+  name: string;
+  key: string;
+  sortable?: boolean;
+};
+
+type sortProps = { name: string; direction: "asc" | "desc" };
+
+export type ControlledTableProps<Row extends BaseRow> = {
+  columns: ColumnItem[];
+  rows: Row[];
+  rowRenderer: (row: Row) => React.ReactNode;
+  tableProps?: React.HTMLAttributes<HTMLTableElement> & { className: string };
+  sortColumn?: sortProps;
+  onSort?: (props: sortProps) => void;
+  onPageChange?: (page: number) => void;
+  page?: number;
+  totalItems?: number;
+  ifEmpty?: React.ReactNode;
+  cols?: React.ReactNode;
+};
+
+/**
+ * A basic reusable table component that supports client-side sorting by a column
+ *
+ * @param columns     - an array of objects with name and key properties
+ * @param rows        - an array of objects with keys that match the column keys
+ * @param rowRenderer - a function that takes a row object and returns a <tr> element
+ * @param tableProps  - additional props to pass to the <table> element
+ */
+
+export function ControlledTable<Row extends BaseRow>({
+  columns,
+  rows,
+  rowRenderer,
+  sortColumn,
+  onSort,
+  page,
+  onPageChange,
+  totalItems,
+  ifEmpty,
+  cols,
+  ...rest
+}: ControlledTableProps<Row>) {
+  return (
+    <div>
+      <table {...rest} className={CS.Table}>
+        <colgroup>{cols}</colgroup>
+        <thead dir="ltr">
+          <tr>
+            {columns.map(column => (
+              <th key={String(column.key)}>
+                {onSort && column.sortable !== false ? (
```

Comment:
Decided it probably made sense to assume every column is sortable unless otherwise defined